rusoto/core/src/request.rs

```diff
@@ -514,6 +526,40 @@ where
     }
 }
 
+impl CompressRequestPayload for HttpClient {
+    fn compress(&self, request: &mut SignedRequest, compression_options: &CompressionOptions) {
+        match request.payload {
+            None => return,
+            Some(SignedRequestPayload::Buffer(ref payload)) => {
+                // Don't compress if request payload length size is lesser than min size
+                if let Some(sz) = compression_options.min_payload_size_for_compression{
+                    if payload.len() < sz{
+                        return;
+                    }
+                }
+                let mut encoder = libflate::gzip::Encoder::new(Vec::<u8>::new()).unwrap();
```

Comment:
I know our existing code isn't the best with using `unwrap` and `expect` but we should see how else we can address an error happening here. Ideally we could return an error that would then be logged, but the least we can do is no-op and log something bad happened.